packages/lit-dev-content/site/tutorials/content/working-with-lists/00.md

```diff
@@ -0,0 +1,13 @@
+In this tutorial, we will be exploring different ways of rendering a list of
+items in Lit!
+
+Let's begin by looking a how we might render a list of names from an array.
+We've already set up a custom element with a `names` property that is an
+array of strings.
+
+We can utlize the `map` array prototype method to generate a list item
+for each item in the array within the unordered list (`<ul></ul>`).
+
+```ts
```

Comment:
Tell the user directly to add this too the code and add the filename before the sample:
`### my-element.ts`

Reply:
As I'm thinking about this change, I'm wondering whether this much spoon feeding is the best way to write interactive tutorials. In my former teaching experience, I'd never want to give away the answer but provide enough clues for students to figure out themselves which makes it more memorable. I know during the kickoff you mentioned people are lazy and want braindead solutions but in that case they can just click the solve button to see the solution. And if they really want to just see working code, they can get that in the doc section or the completed playground examples.

Reply:
It might be doable skipping that for this tutorial since there's only one JS file and one HTML file. Other tutorials have multiple files and due to an oddity on how playground-ide works, it sometimes points to weird files

Reply:
Oh, I'm happy to add filenames for snippets but I was more getting at not providing copy-pasteable code. Prompt can say use the `map` prototype method to render an array of lit-html templates that wrap each item of the array in a list item tag, and the reader has to come up with the code based on that. Maybe I'm thinking too much like a learn-how-to-code classroom kind of tutorial where making the learners think seem more important.

Reply:
also works. As long as it's glance-able

Reply:
I think this is a really interesting idea. I think Tour of Go, for example, does things like this--poses a problem and lets you solve it. And I think we could do that--especially with the addition of the Check button. 

It might make sense, for example, to add a separate "Exercise" step (or steps in a longer tutorial) where you give general instructions and let the users work out a solution. (With the backup of the Check/Solve buttons.) 

Another way to do this is by adding suggestions--typically at the end of a step or the end of a tutorial--of things the user could try. For example, the "Extra credit" items at the end of Steve's tooltip tutorial: https://pr740-77daa6d---lit-dev-5ftespv5na-uc.a.run.app/tutorials/tooltip?mods=tutorialCatalog#09

Encouraging people to play with the code + giving that extra reward of figuring it out yourself. As you said, it really helps make the information stick.